### PR TITLE
feat: support profiling of multiple PID

### PR DESCRIPTION
**Pull request by stormshield-guillaumed**
`perf` support passing multiple PID  with the `-p` flag with a comma separated list. For the flamegraph binary, this PR allows you to pass multiple PID, either with multiple `-p` flag, or a single `-p` flag with a comma separated list to mimic the perf CLI.

I tested my changes on Linux only. I checked the dtrace man page on mac and it seems to support multiple PID by passing the `-p` flag multiple times.

### CODE REVIEW

File: src/bin/flamegraph.rs [hunk 1]
```diff
@@ -9,9 +9,9 @@ use flamegraph::Workload;
 #[derive(Debug, Parser)]
 #[clap(version)]
 struct Opt {
-    /// Profile a running process by pid
-    #[clap(short, long)]
-    pid: Option<u32>,
+    /// Profile a running process by pid (comma separated list)
+    #[clap(short, long, value_delimiter(','))]
+    pid: Option<Vec<u32>>,
```
Comment (djc) [inline src/bin/flamegraph.rs:14]: Can we avoid the `Option` wrapper by using the default for that case?

File: src/lib.rs [hunk 2]
```diff
@@ -100,8 +100,10 @@ mod arch {
                 command.args(&c);
             }
             Workload::Pid(p) => {
+                let p = p.into_iter().map(|p| p.to_string()).collect::<Vec<_>>();
```
Comment (djc) [inline src/lib.rs:103]: Should be able to avoid the intermediate allocations here.

**Review by djc — APPROVED** [review]
Nice! Just a few nits...

File: src/bin/flamegraph.rs (hunk 1 above)
Comment (stormshield-guillaumed) [reply src/bin/flamegraph.rs]: Behaviour of `Option<Vec<T>>` and `Vec<T>` is the same for clap, so I just removed the `Option`

File: src/lib.rs (hunk 2 above)
Comment (stormshield-guillaumed) [reply src/lib.rs]: Not a big fan of the new solution, but the `join` method is only for slices of `Borrow<str>`, so I implemented it manually to make it one argument for perf.

**Comment by djc** [comment]
Thanks!

**Comment by stormshield-guillaumed** [comment]
It seems I broke the CI for macOS with my last change. Do I open a new PR ?

**Comment by djc** [comment]
I was a bit too fast in merging it, but already fixed it in #357.

**Pull request merged**